### PR TITLE
add pdf2html

### PR DESCRIPTION
**Pull request by mowolf**
- [x] try hrml2pdf
- [x] try jsPDF

### CODE REVIEW

File: plugins/vue-html2pdf.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import Vue from "vue";
+import VueHtml2pdf from "vue-html2pdf";
+Vue.use(VueHtml2pdf);
```
Comment (fellnerse) [inline plugins/vue-html2pdf.js:3]: do we still need this?

File: components/ChatVisualization/ChatVisualization.vue [hunk 2]
```diff
@@ -92,48 +89,51 @@
...
       showDownloadPopup: false,
+      ego: this.chat.messagesPerPerson[0].name,
     };
   },
   methods: {
+    setEgo(ego) {
+      this.ego = ego;
+    },
+    download() {
+      this.$gtag.event("render-pdf-started", {
```
Comment (fellnerse) [inline components/ChatVisualization/ChatVisualization.vue:108]: https://github.com/SpiritFour/whatsanalyze/wiki/Google-Analytics

File: components/ChatVisualization/ChatVisualization.vue [hunk 3]
```diff
@@ -92,48 +89,51 @@
...
     },
     onError(event) {
+      console.log("approved", event);
+
       console.log("error", event);
     },
-    async download() {
-      this.$gtag.event("donation-download-started", {
+    downloadSample() {
+      this.$gtag.event("render-sample-pdf-started", {
```
Comment (fellnerse) [inline components/ChatVisualization/ChatVisualization.vue:130]: https://github.com/SpiritFour/whatsanalyze/wiki/Google-Analytics

File: components/charts/Results.vue [hunk 4]
```diff
@@ -78,6 +77,11 @@
 </template>
 
 <script>
+// import * as JsPDF from "jspdf";
```
Comment (PKehnel) [inline components/charts/Results.vue:80]: uncommented block  can be removed

File: functions/pdf.js [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+  };
+  const addBranding = function (
+    _logo = logoBlack,
+    addText = true,
+    x = 130,
+    y = 15
+  ) {
+    // logo
+    doc.addImage(_logo, "PNG", x + 50, y, 13, 13);
+    // title50
```
Comment (PKehnel) [inline functions/pdf.js:64]: title 50?

File: functions/pdf.js [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+    doc.setFontSize(50);
+    doc.text(text, x, y);
+    doc.setLineWidth(0.5);
+    doc.setDrawColor(0, 0, 0); // draw red lines
+    doc.line(x, y + 5, x + 100, y + 5); // horizontal line
+  };
+  const drawAuthorBubble = function (author, x, y) {
+    const rgbAuthorColor = hexToRgb(chat.personColorMap[author].slice(1));
+    doc.setFillColor(rgbAuthorColor[0], rgbAuthorColor[1], rgbAuthorColor[2]);
+    // const author_width = doc.getTextWidth(author);
```
Comment (PKehnel) [inline functions/pdf.js:92]: remove // line

File: functions/pdf.js [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+      factStrings.push("Characters in longest Message: " + fact.longestMessage);
+
+      doc.text(factStrings, marginLeft, usedYSpace + 15);
+
+      usedYSpace += funFactHeight;
+    }
+  });
+
+  // // subtitle
+  // doc.setFontSize(16);
```
Comment (PKehnel) [inline functions/pdf.js:164]: more uncomented code

File: functions/pdf.js [hunk 8]
```diff
@@ -0,0 +1,275 @@
...
+    const numLines = splitMessage.length;
+    const messageHeight = lineHeight * numLines;
+
+    let messageY = calcMessageBodyHeight(numLines); // get start Y Coordinate of Message
+    let messageX = isEgo ? 58 : marginLeft + paddingMessage; // get start X Coordinate of Message
+
+    // Draw bubble
+    if (isSystem) {
+      messageX = 40;
+      doc.setFillColor(53, 53, 38);
```
Comment (PKehnel) [inline functions/pdf.js:194]: maybe we can import colors from a file, so I know what color this is?

Comment (PKehnel) [reply functions/pdf.js:194]: as with person color

File: functions/utils.js [hunk 9]
```diff
@@ -9,6 +9,8 @@ export function downloadBase64File(contentBase64, fileName) {
   document.body.removeChild(downloadLink);
 }
 
+const zeroPad = (num, places) => String(num).padStart(places, "0");
+
 export function getDateString(date, includeTime = true) {
   const months = [
```
Comment (PKehnel) [inline functions/utils.js:15]: can be replaced with moments.month()

**Review by PKehnel — CHANGES_REQUESTED** [review]

File: functions/pdf.js (hunk 8 above)
Comment (mowolf) [reply functions/pdf.js:194]: nah this is a constant color

**Review by PKehnel — APPROVED** [review]

**Pull request merged**